File: String_sol/Roman to integer/Roman_to_integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        val = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]
        sym = ["M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"]

        total = 0
        i = 0
        for i in range(len(sym)):
            while s.startswith[sym[i]]:
                val += val[i]
                s = s[len(sym[i]):]
            i += 1
        return total

File: String_sol/Roman to integer/Roman_to_integer.cpp
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    int romanToInt(string s) {
        vector<int> val{1000,900,500,400,100,90,50,40,10,9,5,4,1};
        vector<string> sym{"M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"};

        int total = 0;
        for (int i = 0; i < sym.size(); i++) {
            while (s.find(sym[i]) == 0) { // string starts with sym[i]
                total += val[i];
                s = s.substr(sym[i].size());
            }
        }
        return total;
    }
};
